src/engine/sidechain/sidechainworker.h

```diff
@@ -2,13 +2,75 @@
 #define SIDECHAINWORKER_H
 
 #include "util/types.h"
+#include "util/fifo.h"
+
+/*
+ * States:
+ * Error        Something errornous has happenned and can't go on
+ * Unknown      First state before init
+ * Init         Initing state don't feed anything in this state
+ * Waiting      Waiting something not ready yet
+ * Busy         Is busy doing something can't process anything new
+ * Ready        Functioning ok
+ * Reading      Reading something and can't do anything else
+ * Writing      Writing something and can't do anything else
+ * Connected    Is connected to storage or server
+ * Connecting   Trying to connect storate or server
+ * Disconnected Ain't connected to storage or server
+ * 
+ * First state should be SIDECHAINWORKER_STATE_UNKNOWN and
+ * if state handling ain't supported by SideChainWorker-class
+ * then SIDECHAINWORKER_STATE_UNKNOWN should be treated as
+ * SIDECHAINWORKER_STATE_READY. Newly written SideChainWorker-class
+ * should support state handling at leas this SIDECHAINWORKER_STATE_READY state.
+ */
+
+enum SidechaingStates {
+    SIDECHAINWORKER_STATE_ERROR = -1,
+    SIDECHAINWORKER_STATE_UNKNOWN,
```

Comment:
Unknown is misleading since the state is known. I usually call such state "vergin".